### PR TITLE
Allow specifying additional arguments for devappserver

### PR DESCRIPTION
**Pull request by briandealwis**
Allows configuring specific settings as per #1999.  No validation of the arguments is performed.  Caveat user!

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -143,7 +143,7 @@
       <!-- bundle: com.google.cloud.tools.appengine -->
       <groupId>com.google.cloud.tools</groupId>
       <artifactId>appengine-plugins-core</artifactId>
-      <version>0.3.5</version>
+      <version>0.3.6-SNAPSHOT</version>
```
Comment (chanseokoh) [inline pom.xml]: No SNAPSHOT. BTW, 0.3.6 is live: http://repo1.maven.org/maven2/com/google/cloud/tools/appengine-plugins-core/0.3.6/

**Comment by chanseokoh** [comment]
We need to update also the appengine-plugins-core version in some MANIFEST files.

```
plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/META-INF/MANIFEST.MF:Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.3.5",
plugins/com.google.cloud.tools.eclipse.appengine.facets/META-INF/MANIFEST.MF:Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.3.5",
plugins/com.google.cloud.tools.eclipse.appengine.localserver/META-INF/MANIFEST.MF:Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.3.5",
plugins/com.google.cloud.tools.eclipse.appengine.newproject/META-INF/MANIFEST.MF:Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.3.5",
plugins/com.google.cloud.tools.eclipse.appengine.standard.java8.test/META-INF/MANIFEST.MF:Import-Package: com.google.cloud.tools.appengine;version="0.3.5",
plugins/com.google.cloud.tools.eclipse.appengine.standard.java8/META-INF/MANIFEST.MF:Import-Package: com.google.cloud.tools.appengine;version="0.3.5",
plugins/com.google.cloud.tools.eclipse.bugreport.test/META-INF/MANIFEST.MF:Import-Package: com.google.cloud.tools.appengine.cloudsdk.serialization;version="0.3.5",
plugins/com.google.cloud.tools.eclipse.bugreport/META-INF/MANIFEST.MF:Import-Package: com.google.cloud.tools.appengine.api;version="0.3.5",
plugins/com.google.cloud.tools.eclipse.bugreport/META-INF/MANIFEST.MF: com.google.cloud.tools.appengine.cloudsdk;version="0.3.5",
plugins/com.google.cloud.tools.eclipse.bugreport/META-INF/MANIFEST.MF: com.google.cloud.tools.appengine.cloudsdk.serialization;version="0.3.5",
plugins/com.google.cloud.tools.eclipse.sdk.ui/META-INF/MANIFEST.MF: com.google.cloud.tools.appengine;bundle-version="0.3.5",
plugins/com.google.cloud.tools.eclipse.sdk/META-INF/MANIFEST.MF:Require-Bundle: com.google.cloud.tools.appengine;bundle-version="0.3.5",
```

**Comment by briandealwis** [comment]
Will do

**Comment by briandealwis** [comment]
Unsure why tests only just broke but /shrug
And added explicit test for program arguments.
PTAL @chanseokoh

**Comment by patflynn** [comment]
did the UI already have an arguments field that was being ignored?

**Comment by briandealwis** [comment]
> did the UI already have an arguments field that was being ignored?

Yes, and it lead to some confusion (#2088).

**Pull request merged**